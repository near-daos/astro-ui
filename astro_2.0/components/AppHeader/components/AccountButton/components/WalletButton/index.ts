export * from 'astro_2.0/components/AppHeader/components/AccountButton/components/WalletButton/WalletButton';
