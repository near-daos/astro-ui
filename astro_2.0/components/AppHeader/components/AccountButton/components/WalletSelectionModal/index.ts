export * from 'astro_2.0/components/AppHeader/components/AccountButton/components/WalletSelectionModal/WalletSelectionModal';
