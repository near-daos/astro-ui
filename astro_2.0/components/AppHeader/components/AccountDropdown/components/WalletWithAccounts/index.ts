export * from 'astro_2.0/components/AppHeader/components/AccountDropdown/components/WalletWithAccounts/WalletWithAccounts';
