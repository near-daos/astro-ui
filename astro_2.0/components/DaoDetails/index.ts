export * from './DaoDetailsMinimized';
export * from './DaoDetailsGrid';
