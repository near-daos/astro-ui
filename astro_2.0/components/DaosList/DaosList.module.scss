@import 'styles/mixins';

.root {
  display: grid;
  grid-row-gap: 24px;
  grid-template-areas:
    'header'
    'content';
  grid-template-columns: 1fr;
  padding-top: 56px;
}

.header {
  display: flex;
  flex-direction: column;
  grid-area: header;

  @include tablet-portrait {
    align-items: center;
    flex-direction: row;
  }

  .title {
    margin: 0 0 20px;

    @include tablet-portrait {
      margin: 0 48px 0 0;
    }
  }
}

.content {
  grid-area: content;
}
