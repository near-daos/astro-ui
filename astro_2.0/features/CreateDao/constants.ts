export const STEPS = {
  INFO: 'info',
  KYC: 'kyc',
  LINKS: 'links',
  MEMBERS: 'members',
  PROPOSALS: 'proposals',
  ASSETS: 'assets',
};
