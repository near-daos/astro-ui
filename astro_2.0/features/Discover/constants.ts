import { Interval } from 'services/DaoStatsService/types';

export const CONTRACT = { contract: 'astro' };

export enum DaoStatsTopics {
  GENERAL_INFO = 'generalInfo',
  USERS_AND_ACTIVITY = 'usersAndActivity',
  GOVERNANCE = 'governance',
  FLOW = 'flow',
  TVL = 'tvl',
  TOKENS = 'tokens',
}

export enum GeneralInfoTabs {
  ACTIVITY = 'activity',
  ACTIVE_DAOS = 'activeDaos',
}

export enum UsersAndActivityTabs {
  ACTIVE_USERS = 'activeUsers',
  ALL_USERS_ON_PLATFORM = 'allUsersOnPlatform',
  USERS_MEMBERS_OF_DAO = 'usersMembersOfDao',
  AVERAGE_NUMBER_OF_USERS_PER_DAO = 'averageNumberOfUsersPerDao',
  NUMBER_OF_INTERACTIONS = 'numberOfInteractions',
  AVERAGE_NUMBER_OF_INTERACTIONS_PER_DAO = 'averageNumberOfInteractionsPerDao',
  ALL_USERS_PER_DAO = 'allUsersPerDao',
}

export enum GovernanceTabs {
  NUMBER_OF_PROPOSALS = 'numberOfProposals',
  VOTE_THROUGH_RATE = 'voteThroughRate',
  ACTIVE_PROPOSALS = 'activeProposals',
  ACTIVE_VOTES = 'activeVotes',
}

export enum FlowTabs {
  TOTAL_IN = 'totalIn',
  TOTAL_OUT = 'totalOut',
  INCOMING_TRANSACTIONS = 'incomingTransactions',
  OUTGOING_TRANSACTIONS = 'outgoingTransactions',
}

export enum TvlTabs {
  PLATFORM_TVL = 'platformTvl',
  VL_IN_BOUNTIES = 'vlInBounties',
  NUMBER_OF_BOUNTIES = 'numberOfBounties',
  VL_OF_BOUNTIES = 'vlOfBounties',
  TVL = 'tvl',
}

export enum TokensTabs {
  NUMBER_OF_FTS = 'numberOfFts',
  VL_OF_FTS = 'vlOfFts',
  NUMBER_OF_NFTS = 'numberOfNfts',
}

export const intervalOptions = [
  { label: 'daily', value: Interval.DAY },
  { label: 'weekly', value: Interval.WEEK },
  { label: 'monthly', value: Interval.MONTH },
];
