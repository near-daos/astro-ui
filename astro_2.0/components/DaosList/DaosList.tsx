import React, { FC, useCallback } from 'react';
import { useRouter } from 'next/router';
import { useTranslation } from 'next-i18next';

import { Button } from 'components/button/Button';
import { CREATE_DAO_URL } from 'constants/routing';

import { useWalletContext } from 'context/WalletContext';

import { WalletType } from 'types/config';
import styles from './DaosList.module.scss';

interface DaosListProps {
  label: string;
}

export const DaosList: FC<DaosListProps> = ({ label, children }) => {
  const { t } = useTranslation();
  const router = useRouter();
  const { accountId, login } = useWalletContext();

  const handleCreateDao = useCallback(
    () => (accountId ? router.push(CREATE_DAO_URL) : login(WalletType.NEAR)),
    [login, router, accountId]
  );

  return (
    <div className={styles.root}>
      <div className={styles.header}>
        <h1 className={styles.title}>{t(label)}</h1>
        <Button variant="black" size="small" onClick={handleCreateDao}>
          {t('createNewDao')}
        </Button>
      </div>
      <div className={styles.content}>{children}</div>
    </div>
  );
};
