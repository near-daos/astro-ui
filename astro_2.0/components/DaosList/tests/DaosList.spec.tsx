/* eslint-disable @typescript-eslint/ban-ts-comment */
import { render } from 'jest/testUtils';
import { useRouter } from 'next/router';
import { fireEvent } from '@testing-library/dom';

import { CREATE_DAO_URL } from 'constants/routing';

import { useWalletContext } from 'context/WalletContext';

import { DaosList } from 'astro_2.0/components/DaosList';

jest.mock('next/router', () => {
  return {
    useRouter: jest.fn(() => ({})),
  };
});

jest.mock('context/WalletContext', () => {
  return {
    useWalletContext: jest.fn(() => ({})),
  };
});

jest.mock('next-i18next', () => ({
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => {
    return {
      t: (str: string): string => str,
    };
  },
}));

describe('Dao List', () => {
  it('Should render component', () => {
    const { container } = render(<DaosList label="Hello World" />);

    expect(container).toMatchSnapshot();
  });

  it('Should navigate to login if user is not logged', () => {
    const login = jest.fn();

    // @ts-ignore
    useWalletContext.mockImplementation(() => ({ login }));

    const { getByRole } = render(<DaosList label="Hello World" />);

    fireEvent.click(getByRole('button'));

    expect(login).toBeCalled();
  });

  it('Should navigate to "create dao" page for logged user', () => {
    // @ts-ignore
    useWalletContext.mockImplementation(() => ({ accountId: 'accountId' }));

    const router = {
      push: jest.fn(),
    };

    // @ts-ignore
    useRouter.mockImplementation(() => router);

    const { getByRole } = render(<DaosList label="Hello World" />);

    fireEvent.click(getByRole('button'));

    expect(router.push).toBeCalledWith(CREATE_DAO_URL);
  });
});
