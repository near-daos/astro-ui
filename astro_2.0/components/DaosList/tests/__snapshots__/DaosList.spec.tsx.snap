// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dao List Should render component 1`] = `
<div>
  <div
    class="root"
  >
    <div
      class="header"
    >
      <h1
        class="title"
      >
        Hello World
      </h1>
      <button
        class="btn black sizeSmall"
        type="button"
      >
        createNewDao
      </button>
    </div>
    <div
      class="content"
    />
  </div>
</div>
`;
